.devcontainer/Dockerfile

```diff
@@ -1,26 +1,30 @@
-FROM mcr.microsoft.com/devcontainers/base:ubuntu
-COPY environment.yml /opt/environment.yml
-
-# Enable by default the conda environment for all users.
-# The prefix is the one used by the micromamba feature.
-ENV CONDA_PREFIX_JAXSIM=/opt/conda/envs/jaxsim
-RUN echo 'function activate_conda() {' >> /etc/bash.bashrc &&\
-    echo '  eval "$(micromamba shell hook -s bash)"' >> /etc/bash.bashrc &&\
-    echo '  micromamba activate ${CONDA_PREFIX_JAXSIM-$CONDA_PREFIX}' >> /etc/bash.bashrc &&\
-    echo '}' >> /etc/bash.bashrc &&\
-    echo '[[ -x $(which micromamba) && -d ${CONDA_PREFIX_JAXSIM-$CONDA_PREFIX} ]] && activate_conda' >> /etc/bash.bashrc &&\
-    echo '[[ -x $(which micromamba) && ! -x $(which mamba) ]] && alias mamba="$(which micromamba)"' >> /etc/bash.bashrc &&\
-    echo '[[ -x $(which micromamba) && ! -x $(which conda) ]] && alias conda="$(which micromamba)"' >> /etc/bash.bashrc
-
-# Provide libGL.so.1 from the host OS
-RUN sudo apt-get update &&\
-    sudo apt-get install -y --no-install-recommends libgl1 &&\
-    rm -rf /var/lib/apt/lists/*
-
-# The Python extension in VSCode is not able to detect the interpreter installed with micromamba.
-# As documented as follows, we provide a suggestion through an environment variable.
-# https://code.visualstudio.com/docs/python/environments#_where-the-extension-looks-for-environments
-ENV WORKON_HOME=$CONDA_PREFIX_JAXSIM
-
-# Specify the workdir of the container
+FROM ghcr.io/prefix-dev/pixi:latest
+
+COPY . /workspace/jaxsim
```

Comment:
Mmh here the whole repository of JaxSim is copied, and it becomes part of the image. Though, when a `devcontainer` is created e.g. from a PR, another repository is cloned within the container. Are we sure that there's no collision on the folder? Is there any alternative way to just copy the repo in the image for gathering the `pixi.lock`, but then remove it since it will be no longer useful? Also in this case, things should be done in the same RUN (layer) to minimize copies.

Reply:
If I don't copy the folder I get an error during the container creation:


<details><summary>Creation Log</summary>
<p>

```Dockerfile
2024-11-05 15:37:17.876Z: Host information
2024-11-05 15:37:17.876Z: ----------------
2024-11-05 15:37:17.876Z: OS: Ubuntu 22.04.5 LTS (stable release)
2024-11-05 15:37:17.876Z: Image details: https://github.com/github/codespaces-host-images/blob/main/README.md
2024-11-05 15:37:17.876Z: ----------------

=================================================================================
2024-11-05 15:37:17.879Z: Configuration starting...
2024-11-05 15:37:17.885Z: Cloning...

=================================================================================
2024-11-05 15:37:17.908Z: Creating container...
2024-11-05 15:37:17.908Z: Cleaning up docker images...
2024-11-05 15:37:18.393Z: $ devcontainer up --id-label Type=codespaces --workspace-folder /var/lib/docker/codespacemount/workspace/jaxsim --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --log-level trace --log-format json --update-remote-user-uid-default never --mount-workspace-git-root false --omit-config-remote-env-from-metadata --skip-non-blocking-commands --skip-post-create --remove-existing-container --config "/var/lib/docker/codespacemount/workspace/jaxsim/.devcontainer/devcontainer.json" --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --container-session-data-folder /workspaces/.codespaces/.persistedshare/devcontainers-cli/cache --secrets-file /root/.codespaces/shared/user-secrets-envs.json
2024-11-05 15:37:18.581Z: @devcontainers/cli 0.68.0. Node.js v18.20.4. linux 6.5.0-1025-azure x64.
2024-11-05 15:37:19.759Z: $ docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.68.0-1730821039755/Dockerfile-with-features -t vsc-jaxsim-6e00638a1d817797e8792c8fe180ce1433b291d973b1cdeb038fafd3e72b5844 --target dev_containers_target_stage --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label /var/lib/docker/codespacemount/workspace/jaxsim
2024-11-05 15:37:20.109Z: #0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile-with-features
#1 transferring dockerfile: 1.66kB done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for ghcr.io/prefix-dev/pixi:latest
2024-11-05 15:37:20.267Z: #3 ...

#4 [auth] prefix-dev/pixi:pull token for ghcr.io
#4 DONE 0.0s
2024-11-05 15:37:20.416Z: 
#3 [internal] load metadata for ghcr.io/prefix-dev/pixi:latest
2024-11-05 15:37:21.009Z: #3 DONE 1.1s
2024-11-05 15:37:21.118Z: 
#5 [dev_container_auto_added_stage_label 1/4] FROM ghcr.io/prefix-dev/pixi:latest@sha256:c0eea541356fe4a4d625f32aef6ddc0a1d8245e5a25e9f74434d244330df76b0
#5 resolve ghcr.io/prefix-dev/pixi:latest@sha256:c0eea541356fe4a4d625f32aef6ddc0a1d8245e5a25e9f74434d244330df76b0 done
#5 sha256:c0eea541356fe4a4d625f32aef6ddc0a1d8245e5a25e9f74434d244330df76b0 683B / 683B done
2024-11-05 15:37:21.119Z: #5 sha256:29137f0e1c992e08a4674a3e06737f4a2ee5335afc3639d11a9c9c42bc0a5a62 703B / 703B done
#5 sha256:0913091111bddbf3e96b6ce3b59f59e521fa92b9e41bc116f324c7125bed2b73 1.94kB / 1.94kB done
#5 sha256:6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac 0B / 29.54MB 0.1s
#5 sha256:08c9db2de2fb189c5bcb966f29f20f6a7150895774547e2f43ac2966deda1850 0B / 15.64MB 0.1s
2024-11-05 15:37:21.219Z: #5 sha256:6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac 29.54MB / 29.54MB 0.2s
2024-11-05 15:37:21.323Z: #5 sha256:08c9db2de2fb189c5bcb966f29f20f6a7150895774547e2f43ac2966deda1850 14.73MB / 15.64MB 0.3s2024-11-05 15:37:21.324Z: 
2024-11-05 15:37:21.518Z: #5 sha256:6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac 29.54MB / 29.54MB 0.4s done
#5 sha256:08c9db2de2fb189c5bcb966f29f20f6a7150895774547e2f43ac2966deda1850 15.64MB / 15.64MB 0.5s2024-11-05 15:37:21.518Z: 
2024-11-05 15:37:21.631Z: #5 sha256:08c9db2de2fb189c5bcb966f29f20f6a7150895774547e2f43ac2966deda1850 15.64MB / 15.64MB 0.5s done2024-11-05 15:37:21.632Z: 
#5 extracting sha256:6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac 0.1s
2024-11-05 15:37:22.271Z: #5 extracting sha256:6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac 0.7s done2024-11-05 15:37:22.271Z: 
2024-11-05 15:37:22.435Z: #5 extracting sha256:08c9db2de2fb189c5bcb966f29f20f6a7150895774547e2f43ac2966deda1850 0.1s2024-11-05 15:37:22.435Z: 
2024-11-05 15:37:22.698Z: #5 extracting sha256:08c9db2de2fb189c5bcb966f29f20f6a7150895774547e2f43ac2966deda1850 0.2s done
2024-11-05 15:37:23.451Z: #5 DONE 2.4s

#6 [dev_container_auto_added_stage_label 2/4] WORKDIR /workspace/jaxsim
2024-11-05 15:37:23.628Z: #6 DONE 0.0s

#7 [dev_container_auto_added_stage_label 3/4] RUN pixi install --environment test-cpu
2024-11-05 15:37:23.650Z: #7 0.173   × could not find pixi.toml or pyproject.toml which is configured to use pixi
2024-11-05 15:37:23.650Z: #7 0.173 
2024-11-05 15:37:23.666Z: #7 ERROR: process "/bin/sh -c pixi install --environment test-cpu" did not complete successfully: exit code: 12024-11-05 15:37:23.671Z: 
2024-11-05 15:37:23.671Z: ------2024-11-05 15:37:23.671Z: 
2024-11-05 15:37:23.674Z:  > [dev_container_auto_added_stage_label 3/4] RUN pixi install --environment test-cpu:2024-11-05 15:37:23.674Z: 
2024-11-05 15:37:23.674Z: 0.173   × could not find pixi.toml or pyproject.toml which is configured to use pixi2024-11-05 15:37:23.674Z: 
2024-11-05 15:37:23.674Z: 0.173 2024-11-05 15:37:23.674Z: 
2024-11-05 15:37:23.674Z: ------2024-11-05 15:37:23.675Z: 
2024-11-05 15:37:23.675Z: Dockerfile-with-features:82024-11-05 15:37:23.675Z: 
2024-11-05 15:37:23.675Z: --------------------2024-11-05 15:37:23.676Z: 
2024-11-05 15:37:23.676Z:    6 |     2024-11-05 15:37:23.676Z: 
2024-11-05 15:37:23.676Z:    7 |     # Install the dependencies2024-11-05 15:37:23.678Z: 
2024-11-05 15:37:23.678Z:    8 | >>> RUN pixi install --environment test-cpu2024-11-05 15:37:23.678Z: 
2024-11-05 15:37:23.678Z:    9 |     2024-11-05 15:37:23.680Z: 
2024-11-05 15:37:23.680Z:   10 |     # Create the shell-hook bash script to activate the environment2024-11-05 15:37:23.680Z: 
2024-11-05 15:37:23.680Z: --------------------
ERROR: failed to solve: process "/bin/sh -c pixi install --environment test-cpu" did not complete successfully: exit code: 1
2024-11-05 15:37:23.680Z: Stop: Run: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.68.0-1730821039755/Dockerfile-with-features -t vsc-jaxsim-6e00638a1d817797e8792c8fe180ce1433b291d973b1cdeb038fafd3e72b5844 --target dev_containers_target_stage --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label /var/lib/docker/codespacemount/workspace/jaxsim
2024-11-05 15:37:23.682Z: {"outcome":"error","message":"Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.68.0-1730821039755/Dockerfile-with-features -t vsc-jaxsim-6e00638a1d817797e8792c8fe180ce1433b291d973b1cdeb038fafd3e72b5844 --target dev_containers_target_stage --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label /var/lib/docker/codespacemount/workspace/jaxsim","description":"An error occurred building the image."}
2024-11-05 15:37:23.682Z: Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.68.0-1730821039755/Dockerfile-with-features -t vsc-jaxsim-6e00638a1d817797e8792c8fe180ce1433b291d973b1cdeb038fafd3e72b5844 --target dev_containers_target_stage --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label /var/lib/docker/codespacemount/workspace/jaxsim
2024-11-05 15:37:23.683Z:     at FtA (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:466:1933)
2024-11-05 15:37:23.686Z:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-11-05 15:37:23.689Z:     at async Pm (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:465:1896)
2024-11-05 15:37:23.690Z:     at async NH (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:465:610)
2024-11-05 15:37:23.690Z: devcontainer process exited with exit code 1
2024-11-05 15:37:23.690Z:     at async YtA (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:482:3771)
2024-11-05 15:37:23.691Z:     at async eB (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:482:4886)
2024-11-05 15:37:23.691Z:     at async prA (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:663:200)
2024-11-05 15:37:23.692Z:     at async drA (/.codespaces/agent/bin/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:662:14706)

====================================== ERROR ====================================
2024-11-05 15:37:23.694Z: Failed to create container.
=================================================================================
2024-11-05 15:37:23.695Z: Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.68.0-1730821039755/Dockerfile-with-features -t vsc-jaxsim-6e00638a1d817797e8792c8fe180ce1433b291d973b1cdeb038fafd3e72b5844 --target dev_containers_target_stage --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label /var/lib/docker/codespacemount/workspace/jaxsim
2024-11-05 15:37:23.695Z: Error code: 1302 (UnifiedContainersErrorFatalCreatingContainer)

====================================== ERROR ====================================
2024-11-05 15:37:23.707Z: Container creation failed.
=================================================================================
2024-11-05 15:37:23.744Z: 

===================================== WARNING ===================================
2024-11-05 15:37:23.744Z: Creating recovery container.
=================================================================================

=================================================================================
2024-11-05 15:37:24.023Z: Creating container...
2024-11-05 15:37:24.061Z: $ devcontainer up --id-label Type=codespaces --workspace-folder /var/lib/docker/codespacemount/workspace/jaxsim --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --log-level trace --log-format json --update-remote-user-uid-default never --mount-workspace-git-root false --omit-config-remote-env-from-metadata --skip-non-blocking-commands --skip-post-create --config "/var/lib/docker/codespacemount/workspace/jaxsim/.devcontainer/devcontainer.json" --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --container-session-data-folder /workspaces/.codespaces/.persistedshare/devcontainers-cli/cache --secrets-file /root/.codespaces/shared/user-secrets-envs.json
2024-11-05 15:37:24.227Z: @devcontainers/cli 0.68.0. Node.js v18.20.4. linux 6.5.0-1025-azure x64.
2024-11-05 15:37:24.574Z: $alpine -c echo Container started
2024-11-05 15:37:24.592Z: Unable to find image 'mcr.microsoft.com/devcontainers/base:alpine' locally
2024-11-05 15:37:24.709Z: alpine: Pulling from devcontainers/base
2024-11-05 15:37:24.752Z: 
[1A[2K
43c4264eed91: Pulling fs layer 
[1B
[1A[2K
f40462439694: Pulling fs layer 
[1B
[1A[2K
78f82a04246d: Pulling fs layer 
[1B
[1A[2K
ab30627c2e5e: Pulling fs layer 
[1B
[1A[2K
1c6e677dc7f6: Pulling fs layer 
[1B
[1A[2K
90218552dd33: Pulling fs layer 
[1B
[1A[2K
43d7c7b4c7f6: Pulling fs layer 
[1B[4A[2K
ab30627c2e5e: Waiting 
[4B[3A[2K
1c6e677dc7f6: Waiting 
[3B[2A[2K
90218552dd33: Waiting 
[2B[1A[2K
43d7c7b4c7f6: Waiting 
[1B2024-11-05 15:37:24.790Z: [7A[2K
43c4264eed91: Downloading  48.34kB/3.624MB
[7B2024-11-05 15:37:24.807Z: [6A[2K
f40462439694: Downloading     409B/409B
[6B2024-11-05 15:37:24.808Z: [6A[2K
f40462439694: Verifying Checksum 
[6B2024-11-05 15:37:24.808Z: [6A[2K
f40462439694: Download complete 
[6B2024-11-05 15:37:24.810Z: [7A[2K
43c4264eed91: Downloading  3.624MB/3.624MB
[7B2024-11-05 15:37:24.810Z: [7A[2K
43c4264eed91: Verifying Checksum 
[7B2024-11-05 15:37:24.811Z: [7A[2K
43c4264eed91: Download complete 
[7B2024-11-05 15:37:24.819Z: [7A[2K
43c4264eed91: Extracting  65.54kB/3.624MB
[7B[5A[2K
78f82a04246d: Downloading     134B/134B
[5B2024-11-05 15:37:24.822Z: [5A[2K
78f82a04246d: 2024-11-05 15:37:24.823Z: Verifying Checksum 
[5B[5A[2K2024-11-05 15:37:24.825Z: 
78f82a04246d: Download complete 
[5B2024-11-05 15:37:24.858Z: [4A[2K
ab30627c2e5e: 2024-11-05 15:37:24.859Z: Downloading     223B/223B
[4B[4A[2K
ab30627c2e5e: Verifying Checksum 
[4B[4A[2K
ab30627c2e5e: Download complete 
[4B2024-11-05 15:37:24.875Z: [3A[2K
1c6e677dc7f6: 2024-11-05 15:37:24.876Z: Downloading     234B/234B
[3B[3A[2K
1c6e677dc7f6: Verifying Checksum 
[3B[3A[2K
1c6e677dc7f6: Download complete 
[3B2024-11-05 15:37:24.904Z: [2A2024-11-05 15:37:24.906Z: [2K2024-11-05 15:37:24.907Z: 
2024-11-05 15:37:24.908Z: 90218552dd33: 2024-11-05 15:37:24.910Z: Downloading  538.9kB/228.2MB
2024-11-05 15:37:24.911Z: [2B2024-11-05 15:37:24.980Z: [7A2024-11-05 15:37:24.983Z: [2K2024-11-05 15:37:24.984Z: 
2024-11-05 15:37:24.985Z: 43c4264eed91: 2024-11-05 15:37:24.986Z: Extracting  2.228MB/3.624MB
2024-11-05 15:37:24.987Z: [7B2024-11-05 15:37:24.989Z: [1A2024-11-05 15:37:24.993Z: [2K2024-11-05 15:37:24.994Z: 
2024-11-05 15:37:24.995Z: 43d7c7b4c7f6: 2024-11-05 15:37:24.996Z: Downloading  440.6kB/43.39MB
2024-11-05 15:37:24.997Z: [1B2024-11-05 15:37:24.998Z: [7A2024-11-05 15:37:24.999Z: [2K2024-11-05 15:37:25.000Z: 
2024-11-05 15:37:25.002Z: 43c4264eed91: 2024-11-05 15:37:25.003Z: Extracting  3.624MB/3.624MB
2024-11-05 15:37:25.004Z: [7B2024-11-05 15:37:25.018Z: [2A2024-11-05 15:37:25.020Z: [2K2024-11-05 15:37:25.021Z: 
2024-11-05 15:37:25.022Z: 90218552dd33: 2024-11-05 15:37:25.023Z: Downloading  11.88MB/228.2MB
2024-11-05 15:37:25.024Z: [2B2024-11-05 15:37:25.029Z: [1A2024-11-05 15:37:25.031Z: [2K2024-11-05 15:37:25.033Z: 
2024-11-05 15:37:25.035Z: 43d7c7b4c7f6: 2024-11-05 15:37:25.037Z: Downloading  10.62MB/43.39MB
2024-11-05 15:37:25.038Z: [1B2024-11-05 15:37:25.066Z: [7A[2K
43c4264eed91: Pull complete 
2024-11-05 15:37:25.067Z: [7B2024-11-05 15:37:25.074Z: [6A[2K
f40462439694: 2024-11-05 15:37:25.074Z: Extracting     409B/409B
[6B2024-11-05 15:37:25.075Z: [6A[2K
f40462439694: Extracting     409B/409B
[6B2024-11-05 15:37:25.114Z: [2A[2K
90218552dd33: Downloading  35.13MB/228.2MB
[2B2024-11-05 15:37:25.134Z: [1A[2K
2024-11-05 15:37:25.135Z: 43d7c7b4c7f6: Downloading  29.64MB/43.39MB
[1B2024-11-05 15:37:25.216Z: [2A[2K
90218552dd33: Downloading  54.05MB/228.2MB
[2B2024-11-05 15:37:25.228Z: [1A[2K
43d7c7b4c7f6: Verifying Checksum 
[1B2024-11-05 15:37:25.229Z: [1A[2K
43d7c7b4c7f6: Download complete 
[1B2024-11-05 15:37:25.317Z: [2A[2K
90218552dd33: Downloading     80MB/228.2MB
[2B2024-11-05 15:37:25.418Z: [2A2024-11-05 15:37:25.418Z: [2K
90218552dd33: Downloading    106MB/228.2MB
[2B2024-11-05 15:37:25.519Z: [2A2024-11-05 15:37:25.519Z: [2K
90218552dd33: Downloading  128.7MB/228.2MB
[2B2024-11-05 15:37:25.620Z: [2A2024-11-05 15:37:25.620Z: [2K
90218552dd33: Downloading  155.7MB/228.2MB
[2B2024-11-05 15:37:25.708Z: [6A[2K
f40462439694: Pull complete 
2024-11-05 15:37:25.708Z: [6B2024-11-05 15:37:25.714Z: [5A[2K
78f82a04246d: 2024-11-05 15:37:25.714Z: Extracting     134B/134B
[5B[5A[2K
78f82a04246d: Extracting     134B/134B
[5B2024-11-05 15:37:25.736Z: [2A[2K
90218552dd33: Downloading  180.6MB/228.2MB
[2B2024-11-05 15:37:25.822Z: [2A[2K
90218552dd33: Downloading  201.7MB/228.2MB
[2B2024-11-05 15:37:25.921Z: [2A2024-11-05 15:37:25.921Z: [2K
90218552dd33: Verifying Checksum 
[2B2024-11-05 15:37:25.922Z: [2A[2K
90218552dd33: Download complete 
[2B2024-11-05 15:37:26.895Z: [5A[2K
78f82a04246d: Pull complete 
[5B2024-11-05 15:37:26.898Z: [4A2024-11-05 15:37:26.898Z: [2K
ab30627c2e5e: Extracting     223B/223B
[4B[4A2024-11-05 15:37:26.899Z: [2K
ab30627c2e5e: Extracting     223B/223B
[4B2024-11-05 15:37:27.107Z: [4A[2K
ab30627c2e5e: Pull complete 
[4B2024-11-05 15:37:27.109Z: [3A2024-11-05 15:37:27.109Z: [2K
1c6e677dc7f6: Extracting     234B/234B
[3B2024-11-05 15:37:27.110Z: [3A[2K
1c6e677dc7f6: Extracting     234B/234B
[3B2024-11-05 15:37:27.128Z: [3A2024-11-05 15:37:27.129Z: [2K
1c6e677dc7f6: Pull complete 
[3B2024-11-05 15:37:27.140Z: [2A[2K
90218552dd33: 2024-11-05 15:37:27.140Z: Extracting  557.1kB/228.2MB
[2B2024-11-05 15:37:27.244Z: [2A[2K
90218552dd33: Extracting  6.128MB/228.2MB
2024-11-05 15:37:27.245Z: [2B2024-11-05 15:37:27.350Z: [2A[2K
90218552dd33: Extracting  13.93MB/228.2MB
2024-11-05 15:37:27.351Z: [2B2024-11-05 15:37:27.455Z: [2A[2K
90218552dd33: 2024-11-05 15:37:27.455Z: Extracting  20.61MB/228.2MB
[2B2024-11-05 15:37:27.556Z: [2A[2K
90218552dd33: Extracting  22.28MB/228.2MB
[2B2024-11-05 15:37:27.680Z: [2A[2K
90218552dd33: Extracting   23.4MB/228.2MB
[2B2024-11-05 15:37:27.803Z: [2A[2K
90218552dd33: Extracting  24.51MB/228.2MB
[2B2024-11-05 15:37:27.920Z: [2A[2K
90218552dd33: Extracting  28.41MB/228.2MB
[2B2024-11-05 15:37:28.021Z: [2A[2K
90218552dd33: Extracting  36.21MB/228.2MB
[2B2024-11-05 15:37:28.130Z: [2A[2K
90218552dd33: Extracting  42.89MB/228.2MB
[2B2024-11-05 15:37:28.244Z: [2A[2K
90218552dd33: 2024-11-05 15:37:28.244Z: Extracting  49.58MB/228.2MB
[2B2024-11-05 15:37:28.347Z: [2A[2K
90218552dd33: Extracting  54.03MB/228.2MB
[2B2024-11-05 15:37:28.458Z: [2A2024-11-05 15:37:28.458Z: [2K
90218552dd33: Extracting  58.49MB/228.2MB
[2B2024-11-05 15:37:28.562Z: [2A[2K
90218552dd33: Extracting  71.86MB/228.2MB
[2B2024-11-05 15:37:28.662Z: [2A[2K
90218552dd33: Extracting  85.79MB/228.2MB
[2B2024-11-05 15:37:28.767Z: [2A[2K
90218552dd33: Extracting  99.71MB/228.2MB
[2B2024-11-05 15:37:28.871Z: [2A[2K
90218552dd33: 2024-11-05 15:37:28.871Z: Extracting  110.3MB/228.2MB
[2B2024-11-05 15:37:29.007Z: [2A[2K
90218552dd33: Extracting  114.2MB/228.2MB
[2B2024-11-05 15:37:29.120Z: [2A[2K
90218552dd33: Extracting  117.5MB/228.2MB
[2B2024-11-05 15:37:29.247Z: [2A[2K
90218552dd33: Extracting  119.8MB/228.2MB
[2B2024-11-05 15:37:29.353Z: [2A[2K
90218552dd33: Extracting  128.1MB/228.2MB
[2B2024-11-05 15:37:29.457Z: [2A[2K
90218552dd33: Extracting  135.4MB/228.2MB
[2B2024-11-05 15:37:29.569Z: [2A2024-11-05 15:37:29.570Z: [2K
90218552dd33: Extracting  143.2MB/228.2MB
[2B2024-11-05 15:37:29.682Z: [2A2024-11-05 15:37:29.682Z: [2K
90218552dd33: Extracting  147.1MB/228.2MB
[2B2024-11-05 15:37:29.784Z: [2A2024-11-05 15:37:29.785Z: [2K
90218552dd33: Extracting  154.3MB/228.2MB
[2B2024-11-05 15:37:29.889Z: [2A[2K
90218552dd33: Extracting  161.5MB/228.2MB
[2B2024-11-05 15:37:29.995Z: [2A[2K
90218552dd33: Extracting  169.3MB/228.2MB
[2B2024-11-05 15:37:30.113Z: [2A[2K
90218552dd33: Extracting  174.4MB/228.2MB
[2B2024-11-05 15:37:30.202Z: [2A[2K
90218552dd33: Extracting  180.5MB/228.2MB
[2B2024-11-05 15:37:30.302Z: [2A[2K
90218552dd33: Extracting  188.3MB/228.2MB
[2B2024-11-05 15:37:30.407Z: [2A[2K
90218552dd33: Extracting  196.6MB/228.2MB
[2B2024-11-05 15:37:30.509Z: [2A[2K
90218552dd33: Extracting  204.4MB/228.2MB
[2B2024-11-05 15:37:30.615Z: [2A[2K
90218552dd33: Extracting  212.8MB/228.2MB
[2B2024-11-05 15:37:30.777Z: [2A[2K
90218552dd33: Extracting  219.5MB/228.2MB
[2B2024-11-05 15:37:30.917Z: [2A[2K
90218552dd33: Extracting  221.7MB/228.2MB
[2B2024-11-05 15:37:31.085Z: [2A[2K
90218552dd33: Extracting  223.9MB/228.2MB
[2B2024-11-05 15:37:31.301Z: [2A[2K
90218552dd33: Extracting  225.6MB/228.2MB
[2B2024-11-05 15:37:31.424Z: [2A[2K
90218552dd33: Extracting  226.2MB/228.2MB
[2B2024-11-05 15:37:31.434Z: [2A[2K
90218552dd33: Extracting  228.2MB/228.2MB
[2B2024-11-05 15:37:35.140Z: [2A[2K
90218552dd33: Pull complete 
[2B2024-11-05 15:37:35.154Z: [1A[2K
43d7c7b4c7f6: 2024-11-05 15:37:35.154Z: Extracting  458.8kB/43.39MB
[1B2024-11-05 15:37:35.277Z: [1A[2K
43d7c7b4c7f6: Extracting  4.588MB/43.39MB
[1B2024-11-05 15:37:35.394Z: [1A[2K
2024-11-05 15:37:35.394Z: 43d7c7b4c7f6: Extracting  5.505MB/43.39MB
[1B2024-11-05 15:37:35.559Z: [1A[2K
43d7c7b4c7f6: 2024-11-05 15:37:35.560Z: Extracting  12.39MB/43.39MB
[1B2024-11-05 15:37:35.680Z: [1A[2K
43d7c7b4c7f6: 2024-11-05 15:37:35.681Z: Extracting  15.14MB/43.39MB
[1B2024-11-05 15:37:35.782Z: [1A[2K
43d7c7b4c7f6: 2024-11-05 15:37:35.782Z: Extracting  21.56MB/43.39MB
[1B2024-11-05 15:37:35.893Z: [1A[2K
2024-11-05 15:37:35.893Z: 43d7c7b4c7f6: Extracting  28.44MB/43.39MB
[1B2024-11-05 15:37:36.025Z: [1A[2K
43d7c7b4c7f6: Extracting  29.82MB/43.39MB
[1B2024-11-05 15:37:36.175Z: [1A[2K
43d7c7b4c7f6: Extracting   31.2MB/43.39MB
[1B2024-11-05 15:37:36.299Z: [1A[2K
43d7c7b4c7f6: Extracting  32.57MB/43.39MB
[1B2024-11-05 15:37:36.405Z: [1A[2K
43d7c7b4c7f6: Extracting  35.78MB/43.39MB
[1B2024-11-05 15:37:36.516Z: [1A[2K
43d7c7b4c7f6: Extracting   36.7MB/43.39MB
[1B2024-11-05 15:37:36.640Z: [1A[2K
43d7c7b4c7f6: Extracting  39.91MB/43.39MB
[1B2024-11-05 15:37:36.744Z: [1A[2K
43d7c7b4c7f6: Extracting  40.83MB/43.39MB
[1B2024-11-05 15:37:36.824Z: [1A[2K
43d7c7b4c7f6: Extracting  43.39MB/43.39MB
[1B2024-11-05 15:37:37.757Z: [1A2024-11-05 15:37:37.758Z: [2K
43d7c7b4c7f6: Pull complete 
[1B2024-11-05 15:37:37.762Z: Digest: sha256:b7a03802a7af5b3bc45e567a697ce4747c470296f900fadce6f9bdc02296a386
2024-11-05 15:37:37.765Z: Status: Downloaded newer image for mcr.microsoft.com/devcontainers/base:alpine
2024-11-05 15:37:37.873Z: Container started
2024-11-05 15:37:38.060Z: Outcome: success User: vscode WorkspaceFolder: /workspaces/jaxsim
2024-11-05 15:37:38.068Z: devcontainer process exited with exit code 0

=================================================================================
2024-11-05 15:37:38.657Z: Running blocking commands...
2024-11-05 15:37:38.688Z: $ devcontainer up --id-label Type=codespaces --workspace-folder /var/lib/docker/codespacemount/workspace/jaxsim --mount type=bind,source=/.codespaces/agent/mount/cache,target=/vscode --user-data-folder /var/lib/docker/codespacemount/.persistedshare --container-data-folder .vscode-remote/data/Machine --container-system-data-folder /var/vscode-remote --log-level trace --log-format json --update-remote-user-uid-default never --mount-workspace-git-root false --omit-config-remote-env-from-metadata --skip-non-blocking-commands --expect-existing-container --config "/var/lib/docker/codespacemount/workspace/jaxsim/.devcontainer/devcontainer.json" --override-config /root/.codespaces/shared/merged_devcontainer.json --default-user-env-probe loginInteractiveShell --container-session-data-folder /workspaces/.codespaces/.persistedshare/devcontainers-cli/cache --secrets-file /root/.codespaces/shared/user-secrets-envs.json
2024-11-05 15:37:38.948Z: @devcontainers/cli 0.68.0. Node.js v18.20.4. linux 6.5.0-1025-azure x64.
2024-11-05 15:37:39.188Z: Outcome: success User: vscode WorkspaceFolder: /workspaces/jaxsim
2024-11-05 15:37:39.193Z: devcontainer process exited with exit code 0

=================================================================================
2024-11-05 15:37:39.208Z: Configuring codespace...

=================================================================================
2024-11-05 15:37:39.252Z: Finished configuring codespace.
```

</p>
</details> 

I believe that this is due to the fact that I'm not using a base image from https://github.com/microsoft/vscode-dev-containers, which may include the either the copy of the folder or mounting the repo folder as volume

Reply:
Since pixi is quite simple to install, in my opinion it's worth sticking with the official images from microsoft. But since this is not a feature that we use a lot, feel free to handle it as you prefer. Do not waste too much time on it.

Reply:
The problem with pixi is that it creates the environment in the `.pixi` of the current folder. I have no idea how Codespaces handles that folder if it already exists in the container.

The best configuration would be the following (more similar to conda):

- A `.pixi` folder somewhere in the image without the editable installation of jaxsim
- When a container is created, enable that environment and `pip install -e .` the active revision

Reply:
Yep that should be disabled during the image creation. This is an inconvenience of using pixi for shipping a conda-like environment.

Reply:
Ok it makes sense to me.